modalc-lib/modal-vector.rkt

```diff
@@ -0,0 +1,45 @@
+#lang racket
+
+(require modalc)
+
+(define (modal-vector/c should-apply-ctc? . inner-ctc)
+  (define vector-contract (apply vector/c inner-ctc))
+  (define inner-ctc-proj
+    (contract-late-neg-projection vector-contract))
+  (make-contract
+   #:name `(modal-vector/c ,(contract-name vector-contract))
+   #:late-neg-projection
+   (λ (blame)
+     (define inner-ctc-proj/blame (inner-ctc-proj blame))
+     (λ (val neg-party)
+       (impersonate-vector val
+                           (λ (vec index ret)
+                             (if (should-apply-ctc? val)
+                                 (vector-ref (inner-ctc-proj/blame val neg-party) index)
+                                 ret))
+                           (λ (vec index ret)
+                             (if (should-apply-ctc? val)
+                                 (begin (inner-ctc-proj/blame val neg-party)
+                                        ret)
+                                 ret)))))))
+
```

Comment:
I imagine we also need `modal-vectorof`? Do the benchmarks use `vectorof`? If the benchmarks only use one or the other then OK, we can get away with just that one.